src/DetailsView/components/adhoc-tab-stops-test-view.tsx

```diff
@@ -61,9 +61,55 @@ export interface AdhocTabStopsTestViewProps {
     userConfigurationStoreData: UserConfigurationStoreData;
 }
 
-export const AdhocTabStopsTestView = NamedFC<AdhocTabStopsTestViewProps>(
-    'AdhocTabStopsTestView',
-    props => {
+export type AdhocTabStopsTestViewState = {
+    autoDetectedFailuresDialogEnabled: boolean;
+};
+
+export class AdhocTabStopsTestView extends React.Component<
+    AdhocTabStopsTestViewProps,
```

Comment:
I think the AdHocTabStopsTestView component already has too many responsibilities and we should try to avoid adding more. In this case, can we not just keep the state in the dialog component?

Additionally, instead of having stateful components, this might be a good opportunity to shift more "view" information into the view store (specifically tab-stops-view-store.ts). This would mean the AdhocTabStopsTestView can remain a functional component and so can the new dialog.

Reply:
@waabid Yeah, I can definitely shift the state to the dialog component. How would it work to have the state in the TabStopsViewStore though, since that's in the details view? If we keep the dialog state in a store, I would think we would want to update it based on `AutomatedTabbingResultsCompleted`, which is kept in the background.

Reply:
For now I've switched over to keeping the state as part of the dialog component, let me know if you have any thoughts on moving the state to the store and I can look more into that as well :)

Reply:
This is good as is.

I understand what you mean now; you need the state from the other store data. I guess you could still move the dialog state to the store but the componentDidUpdate and what not still would be required, so the benefit is minimal.